File: algorithms/java/Challenge_Queue_Reconstruction_by_Height.java
class Solution {
    public int[][] reconstructQueue(int[][] people) {
        Arrays.sort(people, Comparator.comparingInt((int[] p) -> -p[0]).thenComparingInt(p -> p[1]));
        List<int[]> retList = new ArrayList<>(); 
        for(int[] person: people){
            retList.add(person[1], person);
        }
        
        return retList.toArray(new int[retList.size()][2]);
    }
}

File: algorithms/python/Challenge_Queue_Reconstruction_by_Height.py
class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        people = sorted(people, key=lambda x: (-x[0], x[1]))
        retList = [];
        
        for person in people:
            retList.insert(person[1], person)
        
        return retList
